tasks/compile.py

```diff
@@ -99,3 +101,22 @@ def _ts_compile(func, optimize=True):
     cmd_string = " ".join(cmd)
     print(cmd_string)
     call(cmd_string, cwd=TS_DIR, shell=True)
+
+def _codegen_user(user, debug):
+    cmd = " ".join([
+        "codegen_func", # Requires to have bin dir in PATH
+        user,
+    ])
+    if debug:
+        print("Calling {}".format(cmd))
+    call(cmd, shell=True)
```

Comment:
This is duplicating functionality that exists in `codegen_for_user`, however, that doesn't expect `codegen_func` to be on the `PATH` as this does (which is a good idea). It currently needs to run in Dockerfiles and set-up scripts, so checks certain dirs for the codegen binaries. Perhaps it could be changed to fall back to the path (i.e. modifying `find_codegen_xxx` to use the path as the first/last resort).

`codegen_for_user` is also a bit long-winded, I prefer `codegen-user`